fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/info/CreateMTMVInfo.java

```diff
@@ -224,7 +228,18 @@ public void analyzeQuery(ConnectContext ctx) {
         }
         getRelation(planner);
         getColumns(plan);
-        analyzePartition(planner);
+        SessionVariable sessionVariable = statementContext.getConnectContext().getSessionVariable();
+        Set<String> tempDisableRules = sessionVariable.getDisableNereidsRuleNames();
+        // Should not make table without data to empty relation, so add disable rules
+        sessionVariable.setDisableNereidsRules(MTMV_PLANER_DISABLE_RULES);
+        try {
+            analyzePartition(planner);
+        } finally {
+            if (!tempDisableRules.isEmpty()) {
```

Comment:
should not add this `if`, always set back

Reply:
OK, I will fix it